### DISCUSSION TITLE
Self-Deployed Try Catch Blocks

### DISCUSSION THREAD

**Discussion by ndelaquintana** [dotnet/csharplang#8551] [Language Ideas]
/* CASE 1
public void Save()
{
    try
    {
        client.Do();
    }
    catch (Exception e) 
    {
        client.Log(e);
    }
    finally
    {
        client.Dispose();
    }
}
public void NewSave() 
    try => client.Do();
    catch (Exception e) => client.Log(e);
    finally => client.Dispose();
*/

/* CASE 2
public void Save()
{
    try
    {
        client.Do();
    }
    catch 
    {
    }
    finally
    {
    }
}
public void NewSave() 
    try => client.Do(); 
*/

/* CASE 3
void Work(Client client)
{
    while (true)
    {
        try
        {
            client.Do();
        }
        catch (Exception e)
        {
        }
    }
}
void NewWork(Client client)
{
    while (true) try => client.Do();
}
*/

**Comment by HaloFour** [comment]
See: #220

Also, it helps to format the code and put it into [code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) for the purposes of readability.
